src/CodeActionUtil.ts

```diff
@@ -25,6 +25,64 @@ export class CodeActionUtil {
         }
         return CodeAction.create(obj.title, edit);
     }
+
+    private rangeToKey(range: Range) {
+        return `${range?.start?.line}:${range?.start?.character},${range?.end?.line}:${range?.end?.character}`;
+    }
+
+    /**
+     * Generate a unique key for each code action, and only keep one of each action
+     */
+    public dedupe(codeActions: CodeAction[]) {
```

Comment:
This function generates a unique key for each code action, and then only returns one of each unique code action. 

The reason I needed to do this was the `OnScopeGetCodeAction` event would produce a code action for each scope for a missing function, but the action was the same "import "somefile.brs".

Reply:
vscode does not dedupe the code actions, so this would result in many identical code actions showing in the UI.

Reply:
That looks fantastically expensive to compute, if the goal is to just to avoid creating multiple quick-fixes.
I'd suggest just deduplicating on the title and making that clear somehow in the API/doc.

Reply:
this was removed.